src/Views/AccountView.vala

```diff
@@ -108,14 +113,26 @@ public class Installer.AccountView : AbstractInstallerView {
             username_entry.text = username;
         });
 
-        username_entry.changed.connect (check_username);
-        pw_entry.changed.connect (check_password);
-        confirm_entry.changed.connect (check_password);
+        username_entry.changed.connect (() => {
+            valid_username = check_username ();
```

Comment:
I'm not sure this is an improvement over the current method of validating. If we are validating anything inside the dialog we should explicitly update the finish button, in this case I think I would prefer to call `validate_form` inside `check_` and `confirm_` methods. The `valid_` fields should just be set inside those methods.

Reply:
That breaks down with checking and confirming the password which must be both be done if the password entry changes, but only confirming must be done if the confirm entry changes. If each method called `validate_form` then every time the password confirmation entry changed, we would validate the form twice.